### PR TITLE
Remove deprecated configuration

### PR DESCRIPTION
**Pull request by GromNaN**
| namespace                | removed     | replaced by        |
|--------------------------|-------------|--------------------|
| `default_commit_options` | `fsync`     | `j`                |
| `default_commit_options` | `safe`      | `w`                |
| `connections.*.options`  | `fsync`     | `journal`          |
| `connections.*.options`  | `slaveOkay` | `readPreference`   |
| `connections.*.options`  | `timeout`   | `connectTimeoutMS` |
| `connections.*.options`  | `wTimeout`  | `wTimeoutMS`       |

Flagged as deprecated in https://github.com/doctrine/DoctrineMongoDBBundle/pull/820

### CODE REVIEW

File: Tests/DependencyInjection/ConfigurationTest.php [hunk 1]
```diff
@@ -276,7 +273,7 @@ public static function provideMergeOptions(): array
         // the "options" array is totally replaced
         $cases[] = [
             [
-                ['connections' => ['default' => ['options' => ['timeout' => 2000]]]],
+                ['connections' => ['default' => ['options' => ['socketTimeoutMS' => 2000]]]],
```
Comment (GromNaN) [inline Tests/DependencyInjection/ConfigurationTest.php:276]: Simply switched to an option that is supported.

File: UPGRADE-5.0.md [hunk 2]
```diff
@@ -14,6 +14,19 @@ UPGRADE FROM 4.x to 5.0
...
 
+## Configuration
+
+Deprecated options have been removed:
+
+| namespace                | removed     | replaced by        |
+|--------------------------|-------------|--------------------|
+| `default_commit_options` | `fsync`     | `j`                |
+| `default_commit_options` | `safe`      | `w`                |
+| `connections.*.options`  | `fsync`     | `journal`          |
```
Comment (GromNaN) [inline UPGRADE-5.0.md:27]: I'm not sure if the `.*.` is the correct notation we use to reference paths with variable parts.

**Comment by franmomu** [comment]
A reminder that when https://github.com/doctrine/DoctrineMongoDBBundle/pull/820 is merged in 5.0.x, we should remove the `@group legacy` from the test as well.

**Comment by GromNaN** [comment]
> A reminder that when #820 is merged in 5.0.x, we should remove the `@group legacy` from the test as well.

Done, after rebase.

**Review by franmomu — APPROVED** [review]

**Pull request merged**